<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1147</width>
    <height>697</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Hack Assembler</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="1">
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="2" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="font">
           <font>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="text">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt; font-weight:600;&quot;&gt;Hack Assembly Source:&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="asmFileLabel">
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QToolButton" name="newAsmSrcButton">
          <property name="toolTip">
           <string>New Hack Assembly source</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../hackassemblereditor.qrc">
            <normaloff>:/icons/new.png</normaloff>:/icons/new.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="openAsmSrcButton">
          <property name="toolTip">
           <string>Open Hack Assembly source</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../hackassemblereditor.qrc">
            <normaloff>:/icons/open.png</normaloff>:/icons/open.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="saveAsmSrcButton">
          <property name="toolTip">
           <string>Save Hack Assembly source</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../hackassemblereditor.qrc">
            <normaloff>:/icons/save.png</normaloff>:/icons/save.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="3">
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_5">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="text">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Reference Binary&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="copyReferenceButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>:/icons/copy.png</normaloff>:/icons/copy.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="openCmpBinaryButton">
          <property name="toolTip">
           <string>Load comparison binary Hack file.</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../hackassemblereditor.qrc">
            <normaloff>:/icons/open.png</normaloff>:/icons/open.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="2">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="text">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;Compiled Binary&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="copyTranslatedButton">
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>:/icons/copy.png</normaloff>:/icons/copy.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="saveHackBinaryButton">
          <property name="toolTip">
           <string>Save assembled Hack binary file.</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../hackassemblereditor.qrc">
            <normaloff>:/icons/save.png</normaloff>:/icons/save.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="1">
       <widget class="QToolButton" name="translateAllButton">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Translate all code at once.</string>
        </property>
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset resource="../hackassemblereditor.qrc">
          <normaloff>:/icons/translate.png</normaloff>:/icons/translate.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="2,2,2,0,1,0">
        <item>
         <widget class="QToolButton" name="resetButton">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="minimumSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Reset code translation.</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../hackassemblereditor.qrc">
            <normaloff>:/icons/reset.png</normaloff>:/icons/reset.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="runPauseButton">
          <property name="minimumSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Start code translation.</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../hackassemblereditor.qrc">
            <normaloff>:/icons/start.png</normaloff>
            <normalon>:/icons/pause.png</normalon>:/icons/start.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="autoRaise">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="nextButton">
          <property name="minimumSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Go forward one step in code translation.</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../hackassemblereditor.qrc">
            <normaloff>:/icons/next.png</normaloff>:/icons/next.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QSlider" name="speedSlider">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>140</width>
            <height>0</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Code translation speed.</string>
          </property>
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>4</number>
          </property>
          <property name="singleStep">
           <number>3</number>
          </property>
          <property name="pageStep">
           <number>1</number>
          </property>
          <property name="sliderPosition">
           <number>2</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="invertedControls">
           <bool>false</bool>
          </property>
          <property name="tickPosition">
           <enum>QSlider::TicksBothSides</enum>
          </property>
          <property name="tickInterval">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="speedLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>36</width>
            <height>0</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="text">
           <string>x1</string>
          </property>
          <property name="textFormat">
           <enum>Qt::PlainText</enum>
          </property>
          <property name="scaledContents">
           <bool>false</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="2" rowspan="2">
       <widget class="QListWidget" name="translatedCode">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Monospace</family>
          <pointsize>11</pointsize>
         </font>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::SingleSelection</enum>
        </property>
       </widget>
      </item>
      <item row="3" column="3" rowspan="2">
       <widget class="QListWidget" name="referenceCode">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <family>Monospace</family>
          <pointsize>11</pointsize>
         </font>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::SingleSelection</enum>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QPlainTextEdit" name="sourceTextEdit">
        <property name="font">
         <font>
          <family>Monospace</family>
          <pointsize>11</pointsize>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true"/>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1147</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_NewAsmSource"/>
    <addaction name="action_OpenAsmSource"/>
    <addaction name="action_SaveAsmSource"/>
    <addaction name="separator"/>
    <addaction name="action_SaveTranslatedBinary"/>
    <addaction name="action_OpenCmpBinary"/>
    <addaction name="separator"/>
    <addaction name="action_Exit"/>
   </widget>
   <widget class="QMenu" name="menu_Run">
    <property name="title">
     <string>&amp;Run</string>
    </property>
    <addaction name="action_TranslateAll"/>
    <addaction name="separator"/>
    <addaction name="action_RunPauseTranslation"/>
    <addaction name="action_StepTranslation"/>
    <addaction name="action_ResetTranslation"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_About"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Run"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="action_OpenAsmSource">
   <property name="text">
    <string>&amp;Open Hack Assembly Source</string>
   </property>
   <property name="toolTip">
    <string>Open Hack Assembly Source</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_SaveAsmSource">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Save Hack Assembly Source</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S, Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="action_OpenCmpBinary">
   <property name="text">
    <string>Open &amp;Comparison Binary</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+O</string>
   </property>
  </action>
  <action name="action_Exit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="action_StepTranslation">
   <property name="icon">
    <iconset resource="../hackassemblereditor.qrc">
     <normaloff>:/icons/next.png</normaloff>:/icons/next.png</iconset>
   </property>
   <property name="text">
    <string>Translate Next &amp;Line</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_RunPauseTranslation">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../hackassemblereditor.qrc">
     <normaloff>:/icons/start.png</normaloff>
     <normalon>:/icons/pause.png</normalon>:/icons/start.png</iconset>
   </property>
   <property name="text">
    <string>Run / Pause &amp;Translation</string>
   </property>
   <property name="iconText">
    <string>Run / Pause Translation</string>
   </property>
   <property name="toolTip">
    <string>Run / Pause Translation</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="action_ResetTranslation">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../hackassemblereditor.qrc">
     <normaloff>:/icons/reset.png</normaloff>:/icons/reset.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Reset Translation</string>
   </property>
   <property name="toolTip">
    <string>Reset Translation</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+R</string>
   </property>
  </action>
  <action name="action_TranslateAll">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="../hackassemblereditor.qrc">
     <normaloff>:/icons/translate.png</normaloff>:/icons/translate.png</iconset>
   </property>
   <property name="text">
    <string>Translate &amp;All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="action_About">
   <property name="text">
    <string>&amp;About</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="action_SaveTranslatedBinary">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Save Translat&amp;ed Binary</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="action_NewAsmSource">
   <property name="text">
    <string>&amp;New Hack Assembly Source</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+N</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="../hackassemblereditor.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>openAsmSrcButton</sender>
   <signal>clicked()</signal>
   <receiver>action_OpenAsmSource</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>24</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveAsmSrcButton</sender>
   <signal>clicked()</signal>
   <receiver>action_SaveAsmSource</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveHackBinaryButton</sender>
   <signal>clicked()</signal>
   <receiver>action_SaveTranslatedBinary</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>780</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openCmpBinaryButton</sender>
   <signal>clicked()</signal>
   <receiver>action_OpenCmpBinary</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1003</x>
     <y>78</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>translateAllButton</sender>
   <signal>clicked()</signal>
   <receiver>action_TranslateAll</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>99</x>
     <y>608</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nextButton</sender>
   <signal>clicked()</signal>
   <receiver>action_StepTranslation</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>416</x>
     <y>608</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newAsmSrcButton</sender>
   <signal>clicked()</signal>
   <receiver>action_NewAsmSource</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>430</x>
     <y>41</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resetButton</sender>
   <signal>clicked()</signal>
   <receiver>action_ResetTranslation</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>30</x>
     <y>644</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>runPauseButton</sender>
   <signal>clicked()</signal>
   <receiver>action_RunPauseTranslation</receiver>
   <slot>trigger()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>76</x>
     <y>644</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
